src/lib/libast/path/pathnative.c

```diff
@@ -30,25 +30,12 @@
 #include <ast.h>
 
 #if __CYGWIN__
-
-extern void	cygwin_conv_to_win32_path(const char*, char*);
+#include <sys/cygwin.h>
 
 size_t
 pathnative(const char* path, char* buf, size_t siz)
 {
-	size_t		n;
-
-	if (!buf || siz < PATH_MAX)
-	{
-		char	tmp[PATH_MAX];
-
-		cygwin_conv_to_win32_path(path, tmp);
-		if ((n = strlen(tmp)) < siz && buf)
-			memcpy(buf, tmp, n + 1);
-		return n;
-	}
-	cygwin_conv_to_win32_path(path, buf);
-	return strlen(buf);
+	return cygwin_conv_path(CCP_POSIX_TO_WIN_A | CCP_ABSOLUTE, path, buf, siz);
```

Comment:
Looking back on this now, this is wrong per the specification in the comments above, vs the documentation of `cygwin_conv_path`.

For cygwin_conv_path, if size is 0, buf may be NULL and it will return the size of buffer required.  If size is not 0, it will return 0 on success and -1 on error (error including `ENOSPC` if size is too small to hold the result).